packages/next/src/server/request/search-params.ts

```diff
@@ -469,6 +475,80 @@ function makeErroringExoticSearchParams(
   return proxiedPromise
 }
 
+/**
+ * This is a variation of `makeErroringExoticSearchParams` that always throws an
+ * error on access, because accessing searchParams inside of `"use cache"` is
+ * not allowed.
+ */
+export function makeErroringExoticSearchParamsForUseCache(
+  workStore: WorkStore
+): Promise<SearchParams> {
+  const cachedSearchParams = CachedSearchParamsForUseCache.get(workStore)
+  if (cachedSearchParams) {
+    return cachedSearchParams
+  }
+
+  const promise = Promise.resolve({})
+
+  const proxiedPromise = new Proxy(promise, {
+    get(target, prop, receiver) {
+      if (Object.hasOwn(promise, prop)) {
+        // The promise has this property directly. we must return it.
+        // We know it isn't a dynamic access because it can only be something
+        // that was previously written to the promise and thus not an underlying searchParam value
+        return ReflectAdapter.get(target, prop, receiver)
+      }
+
+      switch (prop) {
+        // Object prototype
+        case 'hasOwnProperty':
+        case 'isPrototypeOf':
+        case 'propertyIsEnumerable':
+        case 'toString':
+        case 'valueOf':
+        case 'toLocaleString':
+
+        // Promise prototype
+        // fallthrough
+        case 'catch':
+        case 'finally':
+
+        // Common tested properties
+        // fallthrough
+        case 'toJSON':
+        case '$$typeof':
+        case '__esModule': {
+          // These properties cannot be shadowed because they need to be the
+          // true underlying value for Promises to work correctly at runtime
+          return ReflectAdapter.get(target, prop, receiver)
```

Comment:
I think it was an oversight that we didn't convert the other function to use `wellKnownProperties` like we do in most of places. should probably update both